### PR TITLE
Add `assume_valid_target_reached: bool` to `NetRpc::sync_state` 

### LINKED ISSUE #4485: How to determine if the node is in the "looking for assume-valid-target" stage?
#### Is your feature request related to a problem? Please describe.

To better show the synchronization process, we will inform the user that they are in the early synchronization stage, which is 'looking for assume-valid-target'. So we need to know how to get that the ckb node is looking for assume-valid-target

#### Describe the solution you'd like

Now we are using log matching for determination. But the log changes several times. So I want to know if there is a more accurate way to determine this.

#### Describe alternatives you've considered

Could we use the `sync_status` to get the `best_known_block_number `, and check whether the `best_known_block_number` is less than the `assume-valid-target block` number?

### PR DESCRIPTION
**Pull request by eval-exec**
### What problem does this PR solve?

Issue Number: close #4485 

What's Changed:

### Related changes

- Add `assume_valid_target_reached: bool` and `min_chain_work_reached: bool` to `sync_state` RPC
- let `check_assume_valid_target` don't reset `SyncConfig::assume_valid_target` to `None` if the assume target has found in `Snapshot`.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code ci-runs-only: [ quick_checks,linters ]

Side effects

- Performance regression
- Breaking backward compatibility

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Comment by quake** [comment]
could you please add `min_chain_work` field also?

**Comment by eval-exec** [comment]
I think this PR is ready for review.

File: rpc/src/module/net.rs [hunk 1]
```diff
@@ -719,8 +725,21 @@ impl NetRpc for NetRpcImpl {
         let state = chain.shared().state();
         let (fast_time, normal_time, low_time) = state.read_inflight_blocks().division_point();
         let best_known = state.shared_best_header();
+        let min_chain_work = {
+            let mut min_chain_work_500k_u128: [u8; 16] = [0; 16];
+            min_chain_work_500k_u128.copy_from_slice(&MIN_CHAIN_WORK_500K.to_le_bytes()[..16]);
+            u128::from_le_bytes(min_chain_work_500k_u128)
+        };
```
Comment (eval-exec) [inline rpc/src/module/net.rs:732]: `MIN_CHAIN_WORK` is a value of type u256, however, ckb_jsonrpc does not support Uint256, so I need to cast it to u128 first and then convert it to Uint128.

Cc: @quake

**Review by quake — APPROVED** [review]

File: rpc/README.md [hunk 2]
```diff
@@ -6853,6 +6861,10 @@ The overall chain synchronization state of this local node.
 
 * `low_time`: [`Uint64`](#type-uint64) - The download scheduler's time analysis data, the low is the 9/10 of the cut-off point, unit ms
 
+* `min_chain_work`: [`Uint128`](#type-uint128) - The CKB hardcoded MIN_CHAIN_WORK_500K
```
[the remark this answers is not in the document]
Comment (eval-exec) [reply rpc/README.md]: Updated, removed `MIN_CHAIN_WORK_500K` word in rpc docs.

**Review by quake — DISMISSED** [review]

**Review by quake — APPROVED** [review]

**Pull request merged**